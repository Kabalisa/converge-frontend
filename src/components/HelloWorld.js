import React from 'react'
import '../assets/styles/index.scss'
import '../assets/styles/styles.css'

class HelloWorld extends React.Component {
  static initialState = {
    clicks: 0
  }

  state = HelloWorld.initialState

  incrementClicks = () => {
    this.setState(({ clicks }) => ({ clicks: clicks + 1 }))
  }

  resetClicks = () => {
    this.state.clicks > 0 && this.setState(HelloWorld.initialState)
  }

  render () {
    const { clicks } = this.state
    return (
      <div className='hello-world'>
        <h1>Hello MRM</h1>
        <p>Meeting Room project starts here...</p>
        <button className='hello' onClick={this.incrementClicks}>
          Clicked {clicks} times
        </button>
        <button
          onClick={this.resetClicks}
          style={{ marginLeft: 10 }}
          disabled={clicks <= 0}
        >
          Reset Counts
        </button>
      </div>
    )
  }
}

export default HelloWorld
