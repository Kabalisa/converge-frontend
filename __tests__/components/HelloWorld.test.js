import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import HelloWorld from '../../src/components/HelloWorld';

describe('HelloWorld component', () => {
  const wrapper = shallow(<HelloWorld />);

  it('should render without crushing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HelloWorld />, div);
  });

  it('should render correctly in memory', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should respond to button click', () => {
    wrapper.find('.hello').simulate('click');
    wrapper.update();
    expect(wrapper.state().clicks).toEqual(1);
  });

  it('should reset the number of clicks when reset button is clicked', () => {
    wrapper.setState({ clicks: 0 });
    // increment clicks by one
    wrapper.find('.hello').simulate('click');
    expect(wrapper.state().clicks).toEqual(1);
    wrapper.update();
    // decrement  clicks
    wrapper.find('[disabled=false]').simulate('click');
    wrapper.update();
    // check whether clicks are zero
    expect(wrapper.state().clicks).toEqual(0);
  });
});
