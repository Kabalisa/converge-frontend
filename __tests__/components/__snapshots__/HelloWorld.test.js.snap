// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld component should render correctly in memory 1`] = `
<div
  className="hello-world"
>
  <h1>
    Hello MRM
  </h1>
  <p>
    Meeting Room project starts here...
  </p>
  <button
    className="hello"
    onClick={[Function]}
  >
    Clicked 
    0
     times
  </button>
  <button
    disabled={true}
    onClick={[Function]}
    style={
      Object {
        "marginLeft": 10,
      }
    }
  >
    Reset Counts
  </button>
</div>
`;
